mtags/src/main/scala/scala/meta/internal/mtags/OnDemandSymbolIndex.scala

```diff
@@ -53,15 +54,18 @@ final case class OnDemandSymbolIndex(
   override def addSourceJar(jar: AbsolutePath): Unit = tryRun {
     if (sourceJars.addEntry(jar)) {
       FileIO.withJarFileSystem(jar, create = false) { root =>
-        root.listRecursive.foreach {
-          case source if source.isScala =>
-            try {
-              addSourceFile(source, None)
-            } catch {
-              case NonFatal(e) =>
-                onError.lift(IndexError(source, e))
-            }
-          case _ =>
+        try {
+          root.listRecursive.foreach {
+            case source if source.isScala =>
+              try addSourceFile(source, None)
+              catch {
+                case NonFatal(e) => onError.lift(IndexError(source, e))
+              }
+            case _ =>
+          }
+        } catch {
+          case e: NoSuchFileException =>
+            onError(e)
```

Comment:
I think it can be further simplified. This exception is already handled in the `tryRun`, so it seems that there is no need to wrap the `root.listRecursive` in try statement:
We can just leave it at:
```scala
root.listRecursive.foreach {
  case source if source.isScala =>
    try addSourceFile(source, None)
    catch {
      case NonFatal(e) => onError.lift(IndexError(source, e))
    }
  case _ =>
}
```